torch/csrc/jit/interpreter.cpp

```diff
@@ -668,11 +668,16 @@ struct InterpreterStateImpl {
           if (!instructions[pc].debug_location) {
             throw;
           }
-          if (JITException* jit_e = dynamic_cast<JITException *>(&e)) {
-            instructions[pc].debug_location->wrapAndRethrowException(*jit_e, "operation failed in interpreter");
-          } else {
-            instructions[pc].debug_location->wrapAndRethrowException(e, "operation failed in interpreter");
+          // throw correct subtype
```

Comment:
Can we make this just: 
```
std::string msg = instructions[pc].debug_location->wrapErrorMessage(e.what(), "operation failed in interpreter);
if(dynamic_cast<JITException*>(&e) {
  throw JITException(msg);
} else {
  throw std::runtime_error(msg);
}
```
It is much simpler, and doesn't have  bunch of rethrow logic that occurs across different files.